Bottom: bd9239ad86767c4f5034411393f911ca7a5625e2
Top:    6b3fb496565e0fff4391c7e84ad52750307ad95b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-02 21:21:58 +0100

Make rats snap to the center of pads, not the edges


---

diff --git a/src/rats.c b/src/rats.c
index cb3b1d0..818d19a 100644
--- a/src/rats.c
+++ b/src/rats.c
@@ -139,16 +139,8 @@ FindPad (char *ElementName, char *PinNum, ConnectionType * conn, Boolean Same)
 	    conn->ptr2 = &element->Pad[i];
 	    conn->group =
 	      TEST_FLAG (ONSOLDERFLAG, &element->Pad[i]) ? SLayer : CLayer;
-	    if (TEST_FLAG (EDGE2FLAG, &element->Pad[i]))
-	      {
-		conn->X = element->Pad[i].Point2.X;
-		conn->Y = element->Pad[i].Point2.Y;
-	      }
-	    else
-	      {
-		conn->X = element->Pad[i].Point1.X;
-		conn->Y = element->Pad[i].Point1.Y;
-	      }
+      conn->X = (element->Pad[i].Point1.X + element->Pad[i].Point2.X) / 2;
+      conn->Y = (element->Pad[i].Point1.Y + element->Pad[i].Point2.Y) / 2;
 	    break;
 	  }
       if (i == element->PadN)
